src/main/java/com/xceptance/xlt/engine/XltWebClient.java

```diff
@@ -1574,15 +1577,18 @@ private List<CSSStyleRuleImpl> getCssRulesWithUrls(final HtmlPage page)
     {
         final List<CSSStyleRuleImpl> cssRules = new ArrayList<CSSStyleRuleImpl>();
 
-        final Window window = ((Window) page.getEnclosingWindow().getScriptableObject());
-        final HTMLDocument document = ((HTMLDocument) window.getDocument());
-
-        // check all style sheets
-        final StyleSheetList sheets = document.getStyleSheets();
-        for (int i = 0; i < sheets.getLength(); i++)
+        if (isJavaScriptEnabled())
```

Comment:
with 3.7 js support is no longer required to work with style sheets.

At least for this you can hopefully use HtmlPage.getStyleSheets() https://github.com/HtmlUnit/htmlunit/commit/03877bb3ba4e163523198192dd96ab7b37c0c7f2.
Will try to make a new snapshot later today.

Reply:
Cool! When time permits, we will upgrade to 3.7 and see what we can simplify on our end.